test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsTests.cs

```diff
@@ -2043,35 +2043,80 @@ public void LoadSettings_AddsV3ToEmptyConfigFile_OnlyFirstTime()
                 var result = SettingsTestUtils.RemoveWhitespace(@"<?xml version=""1.0"" encoding=""utf-8""?>
 <configuration>
-    <packageSources>
-        <add key=""nuget.org"" value=""https://api.nuget.org/v3/index.json"" protocolVersion=""3"" />
-    </packageSources>
 </configuration>");
 
+                text.Should().NotContain(v3feed);
                 text.Should().Be(result);
+            }
+        }
 
-                var settingsFile = new SettingsFile(Path.Combine(mockBaseDirectory, "TestingGlobalPath"));
+        [Fact]
+        public void LoadSettings_EmptyUserWideConfigFileAndHasTrackerFile_DoNotAddsV3()
+        {
+            using (var mockBaseDirectory = TestDirectory.Create())
+            {
+                // Arrange
+                var config = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+</configuration>";
+
+                var nugetConfigPath = "NuGet.Config";
+                SettingsTestUtils.CreateConfigurationFile(nugetConfigPath, Path.Combine(mockBaseDirectory, "TestingGlobalPath"), config);
+                var trackFilePath = Path.Combine(mockBaseDirectory, "TestingGlobalPath", "nugetorgadd.trk");
+                File.WriteAllText(trackFilePath, "");
 
                 // Act
-                var section = settingsFile.GetSection("packageSources");
-                section.Should().NotBeNull();
-                settingsFile.Remove("packageSources", section.Items.First());
-                settingsFile.SaveToDisk();
-
-                settings = Settings.LoadSettings(
-                                    root: mockBaseDirectory,
-                                    configFileName: null,
-                                    machineWideSettings: null,
-                                    loadUserWideSettings: true,
-                                    useTestingGlobalPath: true);
+                var settings = Settings.LoadSettings(
+                    root: mockBaseDirectory,
+                    configFileName: null,
+                    machineWideSettings: null,
+                    loadUserWideSettings: true,
+                    useTestingGlobalPath: true);
+
                 // Assert
-                text = SettingsTestUtils.RemoveWhitespace(File.ReadAllText(Path.Combine(mockBaseDirectory, "TestingGlobalPath", "NuGet.Config")));
-                result = SettingsTestUtils.RemoveWhitespace(@"<?xml version=""1.0"" encoding=""utf-8""?>
+                var text = SettingsTestUtils.RemoveWhitespace(File.ReadAllText(Path.Combine(mockBaseDirectory, "TestingGlobalPath", "NuGet.Config")));
+                var v3feed = SettingsTestUtils.RemoveWhitespace(@"<add key=""nuget.org"" value=""https://api.nuget.org/v3/index.json"" protocolVersion=""3"" />");
+                var result = SettingsTestUtils.RemoveWhitespace(@"<?xml version=""1.0"" encoding=""utf-8""?>
 <configuration>
 </configuration>");
 
+                text.Should().NotContain(v3feed);
+                text.Should().Be(result);
```

Comment:
Like the previous test, asserting that a result equals a string that does not contain nuget.org is already validating that the the result doesn't contain nuget.org.

Reply:
Thanks. Fixed.